app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java

```diff
@@ -119,83 +143,22 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
                 // If request is cancelled, the result arrays are empty.
                 if (grantResults.length > 0
                     && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    presenter.cameraPermissionGranted(true);
+                    createCameraSource();
                 } else {
-                    presenter.cameraPermissionGranted(false);
+                    Timber.e("No camera permission granted");
+                    DialogInterface.OnClickListener listener = (dialog, id) -> finish();
+
+                    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                    builder.setTitle("Open Event")
+                        .setMessage(R.string.no_camera_permission)
+                        .setPositiveButton(R.string.ok, listener)
+                        .show();
```

Comment:
As discussed, please restore. I don't know how this helps in implementation of the detector or the development of the app. It makes it untestable